sdk/cosmosdb/cosmos/src/request/RequestOptions.ts

```diff
@@ -1,11 +1,11 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
-import { SharedOptions } from "./SharedOptions";
+import { DedicatedGatewayRequestOptions, SharedOptions } from "..";
```

Comment:
Thoughts on why DedicatedGatewayRequestOptions not co-located with SharedOptions?

Reply:
Either way is _technically_ fine, but this creates a circular dependency because `src/request/index.ts` also imports this file. It's best to ensure the import graph is acyclic.

Reply:
If this is causes a circular reference issue, what is the solution for this? 
@jay-most - can we please make sure that the import graph is acyclic?

Reply:
If this is causes a circular reference issue, what is the solution for this? 
@jay-most - can we please make sure that the import graph is acyclic?

Reply:
No, because SharedOptions does not extend nor implements DedicatedGatewayRequestOptions.